src/session.rs

```diff
@@ -3235,6 +3235,31 @@ impl Session {
             Command::SelectionYank => {
                 self.yank_selection();
             }
+            Command::SelectionFlip(dir) => {
+                if let (Mode::Visual(VisualState::Selecting { .. }), Some(s)) =
+                    (self.mode, self.selection)
+                {
+                    let v = self.active_view_mut();
+                    let s = s.abs().bounds();
+
+                    if s.intersects(v.bounds()) {
+                        let s = s.intersection(v.bounds());
+
+                        v.flip(s, dir);
+
+                        self.selection = Some(Selection::from(s));
+                        self.switch_mode(Mode::Visual(VisualState::Pasting));
+                    }
+                    // I think its handy to flip in place for now, hence these commands
+                    // 1., 2. prevent overlap and paste
+                    // 3.     preserve location so you can flip repeatedly w/ hotkey
```

Comment:
```suggestion
```

Reply:
I think if anything, the code is clear and I was confused by the comment :sweat_smile: